gym_bridges/envs/bridges_env.py

```diff
@@ -261,11 +268,96 @@ def reset(self, state=None, gap_count=None):
 
         return self._state.copy()
 
+    def draw_state(
+        self, screen: pygame.Surface, textures: dict[StateType, pygame.Surface]
+    ):
+        x_coordinates = np.arange(0, self.shape[1])
+        y_coordinates = np.arange(0, self.shape[0])
+
+        block_size = renderer_config.BLOCK_SIZE * 0.5
+
+        x_window_coordinates = np.arange(
+            renderer_config.WINDOW_WIDTH // 2 - block_size * self.shape[1] // 2,
+            renderer_config.WINDOW_WIDTH // 2 + block_size * self.shape[1] // 2,
+            block_size,
+        )
+        y_window_coordinates = np.arange(
+            renderer_config.WINDOW_HEIGHT // 2 - block_size * self.shape[0] // 2,
+            renderer_config.WINDOW_HEIGHT // 2 + block_size * self.shape[0] // 2,
+            block_size,
+        )
+
+        for x_coordinate, x_window_coordinate in zip(
+            x_coordinates, x_window_coordinates
+        ):
+            for y_coordinate, y_window_coordinate in zip(
+                y_coordinates, y_window_coordinates
+            ):
+                rect = pygame.Rect(
+                    x_window_coordinate,
+                    y_window_coordinate,
+                    block_size,
+                    block_size,
+                )
+                if (
```

Comment:
I think the following should work give or take:

for state_block, (x_window_coordinate, y_window_coordinate) in zip(self._state.flatten(), itertools.product(x_window_coordinates, y_window_coordinates)):
   rect = pygame.Rect(
                   x_window_coordinate,
                   y_window_coordinate,
                    block_size,
                    block_size,
               )
screen.blit(textures[state_block], rect)